src/components/ArticlePage.jsx

```diff
@@ -11,12 +12,43 @@ function ArticlePage() {
     const [currentArticle, setCurrentArticle,] = useState(null);
     const [commentsOn, setCommentsOn] = useState(false);
     const [currentComments, setCurrentComments] = useState(null);
-    /*use params for useEffect*/
+    const [voteClicked, setVoteClicked] = useState(false);
+    const [currentVotes, setCurrentVotes] = useState(0);
 
-    const commentClick = (e) => {
+
+    const commentClick = () => {
         setCommentsOn(true);
     }
 
+    const updateVotes = () => {
+        if (voteClicked) {
+            setCurrentVotes(currentVotes - 1);
+            updateArticleVotes({ inc_votes: -1 }, article_id).then((data) => {
+            }).catch((error) => {
+                console.log(error)
+                alert('Failed to update votes at this time. Please try again later');
+            })
+        } else {
+            setCurrentVotes(currentVotes + 1);
+            updateArticleVotes({ inc_votes: 1 }, article_id).then((data) => {
+            }).catch((error) => {
+                console.log(error)
+                alert('Failed to update votes at this time. Please try again later');
```

Comment:
Nice 👍 
Good use of optimistic rendering here. Make sure to undo the optimistic upvote in the catch block though :)